assets/__tests__/browser_test.js

```diff
@@ -0,0 +1,67 @@
+import { Browser } from '../js/phoenix_live_view'
+
+describe('Browser', () => {
+  beforeEach(() => {
+    clearCookies()
+  })
+
+  describe('setCookie', () => {
+    it('sets a cookie', () => {
+      Browser.setCookie('apple', 1234)
+      Browser.setCookie('orange', '5678')
+      expect(document.cookie).toContain('apple')
+      expect(document.cookie).toContain('1234')
+      expect(document.cookie).toContain('orange')
+      expect(document.cookie).toContain('5678')
+    })
+  })
+
+  describe('getCookie', () => {
+    it('returns the value for a cookie', () => {
+      document.cookie = 'apple=1234'
+      document.cookie = 'orange=5678'
+      expect(Browser.getCookie('apple')).toEqual('1234')
+    })
+    it('returns an empty string for a non-existant cookie', () => {
+      document.cookie = 'apple=1234'
+      document.cookie = 'orange=5678'
+      expect(Browser.getCookie('plum')).toEqual('')
+    })
+  })
+
+  describe('redirect', () => {
+    const originalWindowLocation = global.window.location
```

Comment:
Sry a bit naive - what is `global.window`?

Reply:
It's Jest-speak for the window object. `window` is globally available in Jest because it's attached to the `global` object. We could probably refer to it as `window.location`, but the current statement is the most canonical way to reference it in a Jest/JSDOM environment.

Reply:
cool.. thx for explaining!